frontend/lib/wallets/Coinbase.ts

```diff
@@ -0,0 +1,86 @@
+import CoinbaseWalletSDK, { CoinbaseWalletProvider } from '@coinbase/wallet-sdk';
+import { ethers } from 'ethers';
+import { getChainIdByNetworkType, getNetworkConfigByType, getNetworkTypeByChainId } from 'auctions-core/src/network';
+import { setSigner } from 'auctions-core/src/signer';
+import { formatToHexWithoutPad } from 'auctions-core/helpers/format';
+import CoinbaseLogo from '~/assets/icons/wallets/coinbase.svg';
+import AbstractWallet from '~/lib/wallets/AbstractWallet';
+
+export default class Coinbase extends AbstractWallet {
+    public static title = 'Coinbase Wallet';
+    public static icon = CoinbaseLogo;
+    public static downloadUrl = 'https://www.coinbase.com/wallet/downloads';
+
+    private static provider?: CoinbaseWalletProvider;
+
+    private addresses?: string[];
+
+    public get address() {
+        if (Array.isArray(this.addresses) && this.addresses[0]) {
+            return this.addresses[0].toLowerCase();
+        }
+    }
+
+    getProvider(network?: string): ethers.providers.JsonRpcProvider {
+        let rpcUrl: string | undefined;
+        try {
+            rpcUrl = getNetworkConfigByType(network).url;
+        } catch {
+            rpcUrl = process.env.RPC_URL;
+        }
```

Comment:
Why did you still decided to fallback into it pure `process.env.RPC_URL`? in case of the network change it can lead to issues

Reply:
I think that's not responsibility of the wallet to initialise `env` variables (otherwise every wallet will have to do so). Instead, 
1. Wallets doesn't need to be initialised at all before user action
2. In case the wallet was previously connected (and we remember that), wallets doesn't need to be set up until network is set up (`wallet/connect` should be called only when network is there)

2nd is the problem outside of the scope of this PR. Or does 1st also applies here?

Reply:
> we are just initializing (there's no switch), so the network parameter is undefined

Can you please provide network provider during the initialization (so not taken from the store, but provided with the argument to the `connect()`)? Your solution would work too, but hacky!